bdb-ip.md

```diff
@@ -0,0 +1,437 @@
+  /transactions/ID returns non-committed (`UNDECIDED`, `BACKLOG`) transactions.
+- Ideally, the inner workings of BDB do not need to be fully understood in
+  order to work with BDB (compare to databases: Many database users aren't
+  experts on the details of the database's inner workings: "What is a BACKLOG,
+  and why should I care?")
+
+
+##### Proposal 2
+
+- /transaction/ID is returning only transactions included in a VALID block
+
+
+Pros:
+
+- Expected behavior (similar to comparable systems: databases and blockchains
+alike)
+- Transaction's status can still be monitored at /status, as is expected
+- Non-persisted transaction cannot be queried from BDB
+- No mental overhead
+- Interface is safe: Users will be making mission-critical assumptions on the
+  /transactions/id endpoint but might miss the API docs part. We want it's
+  default behavior to be as safe as possible
+- Querying for UNDECIDED or BACKLOG is very difficult to achieve without
+  confusing the user (e.g. [transactions can be in multiple
+  states](https://github.com/bigchaindb/bigchaindb/issues/1038#issuecomment-274049790)).
+  Safes us and the user lots of time
+
+Contra:
+
+- Transactions from BACKLOG OR UNDECIDED blocks cannot be queried anymore
+(actually they can using /blocks and /votes)
+
+
+##### Proposal 3
+
+- /transaction/ID is returning only transactions included in a VALID block
+- /transaction/ID?safe=false (where ?safe=true or lack of is default) may
+return transactions from BAcKLOG or UNDECIDED blocks
+
+
+Pros:
+
+- Expected behavior (similar to comparable systems: databases and blockchains
+alike)
+- Non-persisted transaction can be queried from BDB (is that a Pro?)
+- Interface is safe: Users will be making mission-critical assumptions on the
+  /transactions/id endpoint but might miss the API docs part. We want it's
+  default behavior to be as safe as possible
+
+Contra:
+
+- Might be used as a substitute endpoint instead of /status?tx_id
+- Mental overhead
+- Ideally, the inner workings of BDB do not need to be fully understood in
+  order to work with BDB (compare to databases: Many database users aren't
+  experts on the details of the database's inner workings: "What is a BACKLOG,
+  and why should I care?")
+
+
+#### Proposal 4
+
+- /transactions/id gets a ?status flag
+
+Contra:
+
+- Querying for UNDECIDED or BACKLOG is very difficult to achieve without
+  confusing the user (e.g. [transactions can be in multiple
+  states](https://github.com/bigchaindb/bigchaindb/issues/1038#issuecomment-274049790))
+  Safes us and the user lots of time
+- Might be used as a substitute endpoint instead of /status?tx_id
+- Ideally, the inner workings of BDB do not need to be fully understood in
+  order to work with BDB (compare to databases: Many database users aren't
+  experts on the details of the database's inner workings: "What is a BACKLOG,
+  and why should I care?")
+
+
+##### Personal favorite: Proposal 2
```

Comment:
Let me try to argue the problem away:

According to [w3c's status code definition](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html), a 202 has the following properties:

```
The request has been accepted for processing, but the processing has not been completed. The request might or might not eventually be acted upon [...].
The 202 response is intentionally non-committal. Its purpose is to allow a server to accept a request for some other process (perhaps a batch-oriented process that is only run once per day) without requiring that the user agent's connection to the server persist until the process is completed. The entity returned with this response SHOULD include an indication of the request's current status and either a pointer to a status monitor or some estimate of when the user can expect the request to be fulfilled.
```

to highlight:

```
SHOULD include an indication of the request's current status and either a pointer to a status monitor or some estimate of when the user can expect the request to be fulfilled.
```

For a user knowing HTTP status code, there is no unexpected behavior in this case.
We send 202 back and out HTTP API clearly has a **status monitor**.